pandas/core/frame.py

```diff
@@ -6556,7 +6556,8 @@ def _dispatch_frame_op(self, right, func, axis: Optional[int] = None):
         right = lib.item_from_zerodim(right)
         if not is_list_like(right):
             # i.e. scalar, faster than checking np.ndim(right) == 0
-            bm = self._mgr.apply(array_op, right=right)
+            with np.errstate(all="ignore"):
```

Comment:
could just do the np.errstate once at the top of the method?

Reply:
Yes, that is possible (as mentioned in https://github.com/pandas-dev/pandas/pull/40396#issuecomment-797660478). Personally, I think I prefer putting it just around the mgr operate call / array_op calls, but in the end it doesn't matter much (it's not that the other checks here could potentially gives such warning that would be incorrectly suppressed). So either way.

Reply:
im fine with this, we can clean it up later if needbe